src/BridgedRoom.ts

```diff
@@ -176,6 +176,40 @@ export class BridgedRoom {
         this.puppetOwner = opts.puppet_owner;
 
         this.dirty = true;
+
+        // This almost certainly isn't the best place for this
+        this.joinAllSlackUsers();
+    }
+
+    public async joinAllSlackUsers() {
+        const members = await this.getSlackChannelMembership();
+        if (!members) {
+            return;
+        }
+        return Promise.all(members.map(this.joinUserToRoom, this));
+    }
+
+    private async joinUserToRoom(user_id) {
+        if (!this.team) {
+            return;
+        }
+        const ghost = await this.main.getGhostForSlack(user_id, this.team.domain, this.team.id);
+        return ghost.intent.join(this.matrixRoomId);
+    }
+
+    private async getSlackChannelMembership() {
+        if (!this.botClient || !this.slackChannelId) {
+            return;
+        }
+        // TODO: Handle paging here
```

Comment:
It's almost certainly true that if botClient exists, we have a team. I would urge you to just return early if `this.teams` isn't defined. And then I would try store the bot user_id in a `const` at the start and use that const in the filter.